### ISSUE TITLE
Question: Flux Lora seems to be empty

### ISSUE THREAD

**Issue by a-l-e-x-d-s-9** [bghira/SimpleTuner#715]
I've been training a Lora for a few hours (git checkout from 6 hours ago), and it looks like the training is working from wandb. But when I tested the Lora with ComfyUI, it produced 99.9% identical results as Flux without the Lora.
I'm not generating samples during the training, so I don't have an indication of the progress. The main difference in this training is that I'm using an LR of 0.5 with prodigy instead of 1, compared to my previous training.
ComfyUI is loading my old Lora just fine and it is working, and the new training loads without errors too but it doesn't doing anything.
Is it possible that there is an issue with saving Lora that causes an empty Lora saved?
![image](https://github.com/user-attachments/assets/bcfe29c3-0e84-4a64-9549-227e0d49dad3)

**Comment by bghira** [comment]
i think you have to convert the file for comfyUI. but otherwise there hasn't ever been a problem with updating the weights or saving them - i think also that Prodigy is not a great optim for this model, and we've been having the best success with adamw_bf16. I didn't see the model change much with Prodigy.

**Comment by markrmiller** [comment]
What learning rate are you using with adamw? That was the biggest appeal of prodigy for this, not having to figure out a good lr.

**Comment by bghira** [comment]
1e-4 or 5e-5

**Comment by markuryy** [comment]
> i think you have to convert the file for comfyUI

What? That has never been the case for me, as I opened the [issue](https://github.com/comfyanonymous/ComfyUI/issues/4202) for Flux LoRAs in Comfy and they have been working ever since. Unless SimpleTuner secretly switched to saving in XLabs format or something.

>  i think also that Prodigy is not a great optim for this model, and we've been having the best success with adamw_bf16

I would also agree with this. LR depends on the dataset but Prodigy only worked when I switched the resolution to 512px (among other things but the res was the big one). FWIW I've trained 12 LoRAs, 9 of which used Prodigy and only 1 of those ended up being usable (the 512px one)

**Comment by bghira** [comment]
yes 512px res training has best success

**Comment by a-l-e-x-d-s-9** [comment]
@bghira 
My first Lora training with prodigy LR=1 for 900 steps and GRADIENT_ACCUMULATION_STEPS=16, was very successful with the style - using Schnell inference even with strength 1, but with Dev strength above 0.6 Lora was distorting the images horrendously. And overall Schnell with first Lora producing much better results.
I will try to understand what's happening now with training that doesn't seem to change anything with my second Lora...

**Comment by a-l-e-x-d-s-9** [comment]
First training was successful LR=1 prodigy - very good results with Schnell - up to 900 steps, gradient accumulation 16.
Second training was a failure LR=0.5 prodigy - no effect on results Schnell/Dev - up to 400 steps, gradient accumulation 16.
![image](https://github.com/user-attachments/assets/8915edcf-54a6-4ab2-b0d5-8b27d31d4910)

**Comment by bghira** [comment]
any other settings change? you can see on the Runs tab their parameters overlaid

**Comment by a-l-e-x-d-s-9** [comment]
Here are the parameters of good training:
``` log
Git state
git checkout -b "simpletuner-flux-dwindles" bf4bdb729126cdb9f698a510c8296a30a4540d49
Command
/workspace/SimpleTuner/train.py --model_type=lora --pretrained_model_name_or_path=black-forest-labs/FLUX.1-dev --gradient_checkpointing --set_grads_to_none --gradient_accumulation_steps=16 
--resume_from_checkpoint=latest --snr_gamma=5 --data_backend_config=/workspace/multidatabackend.json --num_train_epochs=0 --max_train_steps=1500 --metadata_update_interval=65 
--use_prodigy_optimizer --learning_rate=1.0 --lr_scheduler=constant --seed 21325151 --lr_warmup_steps=0 --output_dir=/workspace/output/ --inference_scheduler_timestep_spacing=trailing 
--training_scheduler_timestep_spacing=trailing --report_to=wandb --allow_tf32 --mixed_precision=bf16 --base_model_precision=fp8-quanto --i_know_what_i_am_doing --lora_rank=32 
--keep_vae_loaded --flux_lora_target=all --flux --train_batch=1 --max_workers=8 --read_batch_size=12 --write_batch_size=32 --caption_dropout_probability=0.1 --torch_num_threads=8 
--image_processing_batch_size=32 --vae_batch_size=4 --validation_prompt="kisliptu style, woman face closeup, simple background, masterpiece, highres, detailed" --num_validation_images=1 
--validation_num_inference_steps=20 --validation_seed=10 --minimum_image_size=1024 --resolution=1024 --validation_resolution=1024x1024 --resolution_type=pixel --checkpointing_steps=15 
--checkpoints_total_limit=30 --validation_steps=5000 --tracker_run_name=simpletuner-flux-dwindles --tracker_project_name=flux-training --validation_guidance=4.5 --validation_guidance_rescale=0.0 
--validation_negative_prompt="deformed, ugly, scary"
```
Here are parameters of bad training:
``` log
Git state
git checkout -b "simpletuner-flux-cerci" 364d7bfbc1a77feebe84290fe0d7a1326c42e3a4
Command
/workspace/SimpleTuner/train.py --model_type=lora --pretrained_model_name_or_path=black-forest-labs/FLUX.1-dev --gradient_checkpointing --set_grads_to_none --gradient_accumulation_steps=16 
--resume_from_checkpoint=latest --snr_gamma=5 --data_backend_config=/workspace/multidatabackend.json --num_train_epochs=0 --max_train_steps=2000 --metadata_update_interval=65 
--use_prodigy_optimizer --learning_rate=0.5 --lr_scheduler=constant --seed 21325152 --lr_warmup_steps=0 --output_dir=/workspace/output/ --inference_scheduler_timestep_spacing=trailing 
--training_scheduler_timestep_spacing=trailing --report_to=wandb --allow_tf32 --mixed_precision=bf16 --base_model_precision=fp8-quanto --i_know_what_i_am_doing --lora_rank=32 
--keep_vae_loaded --flux_lora_target=all --flux --train_batch=1 --max_workers=8 --read_batch_size=12 --write_batch_size=32 --caption_dropout_probability=0.1 --torch_num_threads=8 
--image_processing_batch_size=32 --vae_batch_size=4 --validation_prompt="woman topless, breasts, nipples, simple background, masterpiece, highres, detailed" --num_validation_images=1 
--validation_num_inference_steps=20 --validation_seed=10 --minimum_image_size=1024 --resolution=1024 --validation_resolution=1024x1024 --resolution_type=pixel --checkpointing_steps=20 
--checkpoints_total_limit=30 --validation_steps=5000 --tracker_run_name=simpletuner-flux-cerci --tracker_project_name=flux-training --validation_guidance=6 --validation_guidance_real=1.0 
--validation_guidance_rescale=0.0 --validation_negative_prompt="deformed, ugly, scary"
```

**Comment by a-l-e-x-d-s-9** [comment]
I have been running a new training today with all the updates and Prodigy LR=1.
I also enabled the samples (5 prompts that some of them related to training) and all samples have zero changes in them.
So I'm very confident that the training is just not working.
I will try to use the last version that worked well for me and check that changes in code are causing it - unless there is something new in the settings the tripping me.

**Comment by mhirki** [comment]
You should add --base_model_default_dtype=fp32 and remove --i_know_what_i_am_doing because then it would actually throw an error at you if you don't specify --base_model_default_dtype=fp32 with Prodigy.

**Comment by a-l-e-x-d-s-9** [comment]
@mhirki 
Thank you - I clearly don't know what I'm doing... 😢
I will make the change and try with the latest version.

**Comment by a-l-e-x-d-s-9** [comment]
Ok, I can see that the first and second samples have small differences, so the training is working now. 🥳️
Thank you all for help!
